common/courses/states.ts

```diff
@@ -89,6 +90,51 @@ export async function denyCourse(course: Course, screeningComment: string | null
     logger.info(`Admin denied Course${course.id}) with screening comment: ${screeningComment}`, { courseId: course.id, screeningComment });
 }
 
+/* ------------------ Course Delete ------------- */
+
+export async function canDeleteCourse(course: Course): Promise<Decision> {
+    const subcoursesForCourse = await getSubcoursesForCourse(course.id, false);
+    if (subcoursesForCourse.length == 0) {
+        return { allowed: true, reason: `Course ${course.id} has no subcourses` };
```

Comment:
The decision.reason is supposed to be "machine readable", i.e. a constant string where we can then maintain error message translations in the frontend. So it should rather be { allowed: false, reason: "has-subcourse" }, { allowed: true }